packages/mattermost/src/mixins/createApply.ts

```diff
@@ -2,47 +2,106 @@ import { baseProvisionerType } from '../'
 
 export const createApplyMixin = (base: baseProvisionerType) => class extends base {
 
-    get mattermostPods() { return {
+    get mattermostPreviewPods() {
+        return {
             kind: 'Pod',
             metadata: {
                 namespace: this.serviceNamespace,
                 labels: {
-                    app: 'mattrmost-preview'
+                    app: 'mattermost-preview'
+                }
+            }
+        }
+    }
+
+    get mattermostClusterPods() {
+        return {
+            kind: 'Pod',
+            metadata: {
+                namespace: this.serviceNamespace,
+                labels: {
+                    app: 'mattermost'
                 }
             }
         }
     }
 
     async createApply() {
         await this.ensureServiceNamespacesExist()
-        await this.ensureMattermostInstalled()
+        await this.installMattermostComponents()
         await this.ensureMattermostIsRunning()
     }
 
-    async ensureMattermostInstalled() {
+    async installMattermostComponents() {
+        const namespace = this.serviceNamespace
+        const {
+            name,
+            users,
+            mattermostLicenseSecret,
+            databaseStorageSize,
+            minioStorageSize
+        } = this.spec
 
-        await this.manager.cluster
-            .begin('Install Mattermost services')
-            .list(this.mattermostPods)
-            .do((result, processor) => {
-                if (!result?.object?.items?.length) {
-                    const namespace = this.serviceNamespace
+        let edition = this.manager.document.metadata.labels['system.codezero.io/edition']
+        if (!edition && this.spec.edition) edition = this.spec.edition
+        const isPreview = (edition === 'preview')
 
-                    processor
-                        .addOwner(this.manager.document)
-                        .upsertFile('../../k8s/preview.yaml', { namespace })
-                }
-            })
-            .end()
+        if (isPreview) {
```

Comment:
It doesn't look like you are using `isPreview` anywhere else, so why not just do: 
```suggestion
        if (edition === 'preview') {
```